### PR TITLE
*: Rereintroduce #pyu command as a board option

### PR DESCRIPTION
**Pull request by Chiiruno**
>inb4 60 comment pull request

### CODE REVIEW

File: go/src/meguca/db/config.go [hunk 1]
```diff
@@ -89,28 +89,57 @@ func scanBoardConfigs(r rowScanner) (c config.BoardConfigs, err error) {
...
-			c.Flags, c.NSFW, c.NonLive, c.PosterIDs, c.RbText,
-			c.Created, c.DefaultCSS, c.Title, c.Notice, c.Rules,
-			pq.StringArray(c.Eightball),
-		)
+	return WriteBoardV(tx, c, version)
+}
+
+// WriteBoardV writes a board complete with configurations to the database
+func WriteBoardV(tx *sql.Tx, c BoardConfigs, version int) error {
+	var q squirrel.Sqlizer
```
Comment (bakape) [inline go/src/meguca/db/config.go]: Why excatly are you adding a new function here?

File: go/src/meguca/db/migrations.go [hunk 2]
```diff
@@ -716,17 +716,76 @@ var migrations = []func(*sql.Tx) error{
 	},
 	// Fixes global moderation
 	func(tx *sql.Tx) (err error) {
-		err = WriteBoard(tx, BoardConfigs{
+		err = WriteBoardV(tx, BoardConfigs{
```
Comment (bakape) [inline go/src/meguca/db/migrations.go]: Really, just inline it here.

File: go/src/meguca/db/util.go [hunk 3]
```diff
@@ -85,6 +85,139 @@ func StartTransaction() (*sql.Tx, error) {
...
+	return
+}
+
+// GetPcount retrieves the board's pyu counter atomically
+func GetPcountA(tx *sql.Tx, b string) (c uint64, err error) {
+	r, err := withTransaction(tx, sq.Select("pcount").
+		From("pyu").
+		Where("id = ?", b)).
+		QueryRow()
+	r.Scan(&c)
```
Comment (bakape) [inline go/src/meguca/db/util.go]: You are ignoring any scanning errors and .Scan() can be chained as:
```go
err =  withTransaction(tx, sq.Select("pcount").
		From("pyu").
		Where("id = ?", b)).
		QueryRow().
                Scan(&c)
```

Same for all cases bellow.

File: go/src/meguca/parser/body.go [hunk 4]
```diff
@@ -82,15 +79,16 @@ func ParseBody(body []byte, board string, internal bool) (
 				links = append(links, l)
 			}
 		case '#':
-			if body[lineStart] == '>' { // Ignore hash commands in quotes
+			// Ignore hash commands in quotes, or #pyu/#pcount if board option disabled
+			if body[lineStart] == '>' || (word[1] == 'p' && !pyu) {
```
Comment (bakape) [inline go/src/meguca/parser/body.go]: Index out of bounds, if length of word = 1.

File: go/src/meguca/server/admin.go [hunk 5]
```diff
@@ -262,10 +262,13 @@ func boardConfData(w http.ResponseWriter, r *http.Request) (
 // Handle requests to create a board
 func createBoard(w http.ResponseWriter, r *http.Request) {
 	var msg boardCreationRequest
+
```
Comment (bakape) [inline go/src/meguca/server/admin.go]: Seriously, why are you doing these?

Comment (Chiiruno) [reply go/src/meguca/server/admin.go]: Everything being to close together triggers my autism, since it's less readable IMO.
If you really want them gone, I'll remove them.

File: go/src/meguca/db/util.go (hunk 3 above)
Comment (Chiiruno) [reply go/src/meguca/db/util.go]: I tried that, it caused a compile error since `withTransaction` returns two types.
That's why I had it scan the first variable.

File: go/src/meguca/parser/body.go (hunk 4 above)
Comment (Chiiruno) [reply go/src/meguca/parser/body.go]: I'll add a length check.

File: go/src/meguca/db/migrations.go (hunk 2 above)
Comment (Chiiruno) [reply go/src/meguca/db/migrations.go]: Pre-mature optimization is the root of all evil.
I just realized that was pointless since it calls `db.version` each time anyway.

File: go/src/meguca/db/util.go (hunk 3 above)
Comment (Chiiruno) [reply go/src/meguca/db/util.go]: `go/src/meguca/db/util.go:113:11: multiple-value withTransaction(tx, sq.Select("pcount").From("pyu").Where("id = ?", b)).QueryRow() in single-value context`

Comment (Chiiruno) [reply go/src/meguca/db/util.go]: Also aren't the errors always handled outside of these functions?
If not, what's the point of the `if err != nil {` spam?

Comment (Chiiruno) [reply go/src/meguca/db/util.go]: Never mind I see what you mean now, I need to add the check now since it has it's own variable now and isn't checked for errors.

File: go/src/meguca/server/admin.go (hunk 5 above)
Comment (bakape) [reply go/src/meguca/server/admin.go]: Nah, non-critical.

File: go/src/meguca/db/migrations.go (hunk 2 above)
Comment (bakape) [reply go/src/meguca/db/migrations.go]: Less about premature optimization on more about not defining functions you use once in the global context.

**Comment by bakape** [comment]
All done? If not, ping me, when you are.

File: go/src/meguca/server/admin.go (hunk 5 above)
Comment (Chiiruno) [reply go/src/meguca/server/admin.go]: Let me rephrase that: Do you want me to not do that from now on?

**Comment by Chiiruno** [comment]
@bakape Yeah it's done, unless there's more that you see that needs to be done.

File: go/src/meguca/server/admin.go (hunk 5 above)
Comment (bakape) [reply go/src/meguca/server/admin.go]: Non-critical. I try to not be as opinionated.

**Comment by bakape** [comment]
Please cherry-pick 88c72fe for the added tests. Also, it seems that currently `#pyu` does not show in newly created posts and only after a refresh. Some kind of bug in the client-side parser.

**Comment by bakape** [comment]
Perhaps. Why though? Try digging in with a debugger.

**Comment by Chiiruno** [comment]
Not sure, I may have fixed it with some of the changes to make things more "atomic".
Also I can't into debuggers for the life of me.
"Whatever happened last time is either because of high volume or something I have no idea about."

File: go/src/meguca/parser/body.go [hunk 6]
```diff
@@ -20,14 +21,9 @@ func (e ErrNonPrintable) Error() string {
 	return fmt.Sprintf("contains non-printable character: %d", int(e))
 }
 
-// Needed to avoid cyclic imports for the 'db' package
-func init() {
-	common.ParseBody = ParseBody
```
Comment (bakape) [inline go/src/meguca/parser/body.go]: @Chiiruno 
Well, I found the problem
Causes #741
